lib/galaxy/objectstore/s3.py

```diff
@@ -581,7 +601,7 @@ def _create(self, obj, **kwargs):
 
     def _empty(self, obj, **kwargs):
         if self._exists(obj, **kwargs):
-            return bool(self._size(obj, **kwargs) > 0)
+            return bool(self._size(obj, **kwargs) == 0)
```

Comment:
I think all the other object stores have to same issue. And we could drop the unnecessary bool cast.